### DISCUSSION TITLE
Anyone got voltage envelopes to go with the clock speeds?

### DISCUSSION THREAD

**Discussion by SimonMerrett** [SpenceKonde/DxCore#142] [Q&A] [answered]
I'm keen to play with the Dx chips. About to order a 128DB48 in the extended temperature range. For most other recent AVRs we get a recommended voltage range for the different clock frequencies, such as 
![image](https://user-images.githubusercontent.com/26767525/133632634-ff0ad71e-27e6-4eae-8248-f1a17187e40e.png)
But I can't find anything similar for the Dx chips. Has anyone tested / could someone please test between 1.8 and 3.3V to see what clock speeds are workable in that voltage range at normal room temperatures?

**Comment by SpenceKonde** [comment, marked as answer]
Refer to the marketing hype on microchip website! Voltage has no effect on maximum clock speed (the cpu core, which is what is the limiting factor as far as I've seen (clodk it too high and it'll start doing math wrong, like other AVRs - but on Dx it has a regulator internally and the core is running at 1.x volts). 
So no V dependence. And they also indicate that it works over the whole temperature range at max rated speed. 
What they don't say is that at room temperature, they overclock really well - they all seem to hit 32 MHz, even 40 at room temperature, and I've gotten an extended temp range working at 48 with external clock (which is generally better than a crystal), which didn't work on the non-extended temp range - extended temp range may just be the "top bin". But those are small sample sizes;

**Reply by SimonMerrett** [reply]
Well that's good! Makes a load of circuits running on 2x AA cells or single lithium cells feasible with no regulator. Although the latter needs protection circuitry so not as exciting as the former, in my humble opinion.

**Reply by SimonMerrett** [reply]
Ps, did I miss this hype-worthy fact it in the datasheet or is that sort of information not worthy of inclusion?!

**Reply by steve-chamberlin** [reply]
What supply voltage were your 32, 40, and 48 MHz overclock tests done at? It seems hard to believe there's really no voltage dependence to the overclocking limit. Even if the CPU core has a regulator for 1.8V, wouldn't it make a significant difference in power dissipation whether I/O pins are switching 3.3V or 5V or something else?

**Reply by SpenceKonde** [reply]
My tests were at 5v (well. the 4.7 you get out of USB port + di, and they were not very rigorous or anything - but if you are disipating a measurable amount of heat in the part, you're doing something very wrong. 

You should always test planned overclocking in the condition you plan to use the device under.

**Comment by SpenceKonde** [comment]
It is there, but not all hyped up. I mean it's right there in electrical
charachteristics

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***
